### DISCUSSION TITLE
Shift+Enter for newline in in-game browser textbox

### DISCUSSION THREAD

**Discussion by Gyoshi** [ppy/osu#23715] [Ideas]
I cannot newline in in-game browser using shift+enter. The fact that the hint text that is present on the normal browser is not there makes me think that maybe it is known already, but I could not find a related issue/discussion.

![image](https://github.com/ppy/osu/assets/41743877/16602a84-50a6-4a90-9250-3573402a39ed)

![image](https://github.com/ppy/osu/assets/41743877/cdd3df6f-6d1d-45a0-8a60-bee744ee9bb1)

**Comment by Joehuu** [comment]
> The fact that the hint text that is present on the normal browser is not there makes me think that maybe it is known already, but I could not find a related issue/discussion.

There is only a code comment tracking this:
https://github.com/ppy/osu/blob/d78df0b084c75c3ab608485c425954a2ebbc920b/osu.Game/Overlays/Comments/CommentsContainer.cs#L413-L414

**Reply by peppy** [reply]
The true reason for this is that our textboxes don't support this yet 😅. Although seemingly @frenzibyte may have started on it a few years ago (https://github.com/ppy/osu-framework/issues/3210).

**Reply by frenzibyte** [reply]
I believe I had way too much abstraction back when I attempted this. I'm not super confident in working back on this as the current textbox implementation was quite fragile to simple changes as far as I remember, but if it's a high priority then I can give it a whirl.
